src/pychart/_interpreter/callable.py

```diff
@@ -0,0 +1,24 @@
+from typing import Any, List
+from src.pychart._interpreter.environment import Environment
+
+
+class PychartCallable:
+    @staticmethod
+    def from_expr(expr: Any):
+        if isinstance(expr, PychartCallable):
+            return expr
+        raise RuntimeError("Could not convert to callable.")
+
+    def __call__(self, environment: Environment, args: List[Any]) -> Any:
+        raise RuntimeError("Cannot call base callable.")
+
+    def arity(self) -> int:
+        raise RuntimeError("Cannot call base callable.")
+
+
+class InputFunc(PychartCallable):
+    def __call__(self, environment: Environment, args: List[Any]) -> Any:
+        return input(args[0])
+
+    def arity(self) -> int:
+        return 1
```

Comment:
This is an example of a `Callable` it can be bound to the environment (as a variable), when retrieved (as the expr in the call expr) it will run the `__call__` method. Arity tells how many args.